pinot-controller/src/main/java/org/apache/pinot/controller/tuner/TableConfigTuner.java

```diff
@@ -27,11 +29,18 @@
  * Interface for Table Config Tuner.
  */
 public interface TableConfigTuner {
+  /**
+   *
+   */
   /**
    * Used to initialize underlying implementation with Schema
    * and custom properties (eg: metrics end point)
+   *
+   * @param pinotHelixResourceManager Pinot Helix Resource Manager to access Helix resources
+   * @param tunerConfig Tuner configurations
+   * @param schema Table schema
    */
-  void init(TunerConfig props, Schema schema);
+  void init(@Nullable PinotHelixResourceManager pinotHelixResourceManager, TunerConfig tunerConfig, Schema schema);
```

Comment:
Is the resource manager nullable? If it’s nullable just for the tests, we should rather mock it

Reply:
With resource manager passed in, not sure if we still need the schema

Reply:
No, not for the tests, the existing implementation of `RealTimeAutoIndexTuner` does not use it.

Reply:
As for the schema, I think as we add more tuners, we will figure out the right interface, so I'll keep schema for sometime.

Reply:
It is okay to not use it, but the caller should always pass in the resource manager right?
Let's annotate the interface as evolving if we are still figuring out the right interface

Reply:
The current tuner does not even have access to resource manager. It is a very simplistic tuner if you see the implementation. Will add annotation to the interface.